### PR TITLE
Improve DisplacementAboutAxis

### PR DESCRIPTION
**Pull request by arovinelli**
As suggested by @bwspenc I did small changes to allow `DisplacementAboutAxis` to use an incremental formulation, thus allowing to start the rigid rotation at any time during the simulation

### CODE REVIEW

**Comment by arovinelli** [comment]
@jiangwen84 can you give a look at this, it's pretty quick

**Comment by jiangwen84** [comment]
I know @bwspenc recently discovered that `DisplacementAboutAxis ` needs to be ran on the undisplaced mesh. You made a change here to use incremental form and for that I think you have to use displaced mesh, is that correct?

**Comment by arovinelli** [comment]
@jiangwen84 no actually you don't need to use disaplced_mesh (it would be a nightmare to do it this way). The incremental formulation adds previous solution displacements to the undisplaced node location. Then you apply the rotation as angular velocity rather than total rotation.
Basically you only need to couple in the previous displacemnts nodal solution

**Comment by arovinelli** [comment]
@jiangwen84 now that you point this 

> I know @bwspenc recently discovered that `DisplacementAboutAxis ` needs to be ran on the undisplaced mesh.

Now that I think at it, this BC should error out with `use_displaced_mesh=true`

**Comment by arovinelli** [comment]
> @jiangwen84 now that you point this
> 
> > I know @bwspenc recently discovered that `DisplacementAboutAxis ` needs to be ran on the undisplaced mesh.
> 
> Now that I think at it, this BC should error out with `use_displaced_mesh=true`

but you already set [use_displaced_mesh to false](https://github.com/idaholab/moose/blob/91e607fcf2d3e739ff0639ae77be4bf639f92e43/modules/tensor_mechanics/src/bcs/DisplacementAboutAxis.C#L23)

so we don't need to make other changes

**Review by jiangwen84 — CHANGES_REQUESTED** [review]
Looks good and a few minor comments. 

> @jiangwen84 no actually you don't need to use disaplced_mesh (it would be a nightmare to do it this way). The incremental formulation adds previous solution displacements to the undisplaced node location. Then you apply the rotation as angular velocity rather than total rotation.
> Basically you only need to couple in the previous displacemnts nodal solution

Don't you need to consider the incremental rotation that is due to finite strain when you apply your angular rotation?

File: modules/tensor_mechanics/src/bcs/DisplacementAboutAxis.C [hunk 1]
```diff
@@ -70,15 +81,22 @@ DisplacementAboutAxis::computeQpValue()
...
   p_old(3, 0) = 1;
 
+  if (_angular_velocity)
+    for (unsigned int i = 0; i < _ndisp; i++)
+      p_old(i, 0) += (*_disp_old[i])[_qp];
+
   ColumnMajorMatrix p_new = rotateAroundAxis(p_old, angle);
 
-  return p_new(_component, 0) - p_old(_component, 0);
+  return p_new(_component, 0) - p(_component);
```
[the remark this answers is not in the document]
Comment (arovinelli) [reply modules/tensor_mechanics/src/bcs/DisplacementAboutAxis.C:99]: > For the old case (_angular_velocity = false), looks like you changed p_old -> p, are they the same? I think they are, but I do not understand why they created p_old before. Do you know about that?

The only reason I can give you is that they wanted a 4 element array to work with rotateAroundAxis method. For the old case P_old[0:2] is the same as `p[0:2]` (initial node location).
For the new case, P_old = p + _old_dispalcements, so they are not the same anymore

> 
> Also, can you change p(_component) to p(_component, 0) to be consistent with p_new. I know they are the same for this case.

This prevents me from compiling 
```
error: no match for call to \u2018(libMesh::Point) (const int&, int)\u2019
```

**Comment by arovinelli** [comment]
> Don't you need to consider the incremental rotation that is due to finite strain when you apply your angular rotation?

I don't understand the question. This is BC in which we enforce a displacement value such that a surface will rotate rigidly. So for the nodes on which this BC applies, we forget about strains.
mechanics will take care of that for free nodes

File: modules/tensor_mechanics/src/bcs/DisplacementAboutAxis.C (hunk 1 above)
[the remark this answers is not in the document]
Comment (jiangwen84) [reply modules/tensor_mechanics/src/bcs/DisplacementAboutAxis.C:102]: Ok, I forgot p is Point instead of Matrix.

**Comment by jiangwen84** [comment]
> > Don't you need to consider the incremental rotation that is due to finite strain when you apply your angular rotation?
> 
> I don't understand the question. This is BC in which we enforce a displacement value such that a surface will rotate rigidly. So for the nodes on which this BC applies, we forget about strains.
> mechanics will take care of that for free nodes

Maybe you are right. I was thinking that you might want to rotate based on  deformed configuration.

**Comment by arovinelli** [comment]
> > > Don't you need to consider the incremental rotation that is due to finite strain when you apply your angular rotation?
> > 
> > 
> > I don't understand the question. This is BC in which we enforce a displacement value such that a surface will rotate rigidly. So for the nodes on which this BC applies, we forget about strains.
> > mechanics will take care of that for free nodes
> 
> Maybe you are right. I was thinking that you might want to rotate based on deformed configuration.

Well, you exactly do that by adding the old displacement values to the initial node location (thus recovering the node position at the beginning of the increment). Then, given an axis of rotation and an angular velocity, you compute the new node position at the end of the increment. However, while rotating you can't stretch (you couldn't have done this even before). Basically the major improvement of this PR is allowing rigid rotation after you imposed some arbitrary stretch.

**Comment by arovinelli** [comment]
I fixed this is ready to be merged. Any second thoughts @jiangwen84 and @bwspenc ?

**Comment by arovinelli** [comment]
ping @jiangwen84 and @bwspenc

**Review by bwspenc — APPROVED** [review]
Looks good to me. Thanks for contributing this!

**Comment by bwspenc** [comment]
The tests are being re-run because I submitted a small update with some tweaks to the documentation. I'll merge it once they pass.

**Comment by arovinelli** [comment]
@bwspenc thanks for merging this.
Always happy to contribute!

**Pull request merged**